views/badge.jade

```diff
@@ -0,0 +1,48 @@
+if status === 'success'
+  svg(xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="102" height="20")
+    linearGradient(id="b" x2="0" y2="100%")
+      stop(offset="0" stop-color="#bbb" stop-opacity=".1")
+      stop(offset="1" stop-opacity=".1")
+    clipPath(id="a")
+      rect(width="102" height="20" rx="3" fill="#fff")
+    g(clip-path="url(#a)")
+      path(fill="#555" d="M0 0h49v20H0z")
+      path(fill="#97CA00" d="M49 0h53v20H49z")
+      path(fill="url(#b)" d="M0 0h102v20H0z")
+    g(fill="#fff" text-anchor="middle" font-family="DejaVu Sans,Verdana,Geneva,sans-serif" font-size="11")
+      text(x="24.5" y="15" fill="#010101" fill-opacity=".3") yaydoc
+      text(x="24.5" y="14") yaydoc
+      text(x="74.5" y="15" fill="#010101" fill-opacity=".3") success
+      text(x="74.5" y="14") success
+else if status === 'failed'
+  svg(xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="88" height="20")
+    linearGradient(id="b" x2="0" y2="100%")
+      stop(offset="0" stop-color="#bbb" stop-opacity=".1")
+      stop(offset="1" stop-opacity=".1")
+    clipPath(id="a")
+      rect(width="88" height="20" rx="3" fill="#fff")
+    g(clip-path="url(#a)")
+      path(fill="#555" d="M0 0h49v20H0z")
+      path(fill="#E05d44" d="M49 0h53v20H49z")
+      path(fill="url(#b)" d="M0 0h102v20H0z")
+    g(fill="#fff" text-anchor="middle" font-family="DejaVu Sans,Verdana,Geneva,sans-serif" font-size="11")
+      text(x="24.5" y="15" fill="#010101" fill-opacity=".3") yaydoc
+      text(x="24.5" y="14") yaydoc
+      text(x="67.5" y="15" fill="#010101" fill-opacity=".3") failed
+      text(x="67.5" y="14") failed
+else
+  svg(xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="94" height="20")
+    linearGradient(id="b" x2="0" y2="100%")
+      stop(offset="0" stop-color="#bbb" stop-opacity=".1")
+      stop(offset="1" stop-opacity=".1")
+    clipPath(id="a")
+      rect(width="94" height="20" rx="3" fill="#fff")
+    g(clip-path="url(#a)")
+      path(fill="#555" d="M0 0h49v20H0z")
+      path(fill="#9f9f9f" d="M49 0h45v20H49z")
+      path(fill="url(#b)" d="M0 0h94v20H0z")
+    g(fill="#fff" text-anchor="middle" font-family="DejaVu Sans,Verdana,Geneva,sans-serif" font-size="11")
+      text(x="24.5" y="15" fill="#010101" fill-opacity=".3") yaydoc
+      text(x="24.5" y="14") yaydoc
+      text(x="70.5" y="15" fill="#010101" fill-opacity=".3") invalid
+      text(x="70.5" y="14") invalid
```

Comment:
The three blocks are mostly similar. Can something be done to avoid this code repetition?